src/library/dao/trackdao.cpp

```diff
@@ -1723,7 +1723,7 @@ bool TrackDAO::detectMovedTracks(QSet<TrackId>* pTracksMovedSetOld,
                     "INNER JOIN library ON track_locations.id=library.location "
                     "WHERE track_locations.location IN (%1) AND "
                     "filename=:filename AND "
-                    "duration=:duration").arg(
+                    "ABS(duration - :duration) < 1").arg(
```

Comment:
Please insert a comment what the 1 means, and why it is required.  
Is this a side effect from our repaired engine? https://bugs.launchpad.net/mixxx/+bug/1666275

Do we need to invalidate beatgrid and waveforms?

Reply:
Comment is in place. :smile:

Basically, I believe this is a regression from #970, so I added tolerance of 1 second when searching for the (re)moved track.

> Is this a side effect from our repaired engine? https://bugs.launchpad.net/mixxx/+bug/1666275
> Do we need to invalidate beatgrid and waveforms?

I don't think so, I believe this is a different issue.